### PR TITLE
Mpuncel/env extractor

### PR DESCRIPTION
**Pull request by mpuncel**
This PR depends on #678 and #698 so review those first

### CODE REVIEW

**Comment by anthonybishopric** [comment]
This PR looks like it needs a rebase first

File: pkg/preparer/podprocess/env_extractor.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+// Write exit code and exit status along with some information pulled from the current
+// environment into the configured sqlite database. The environment extractor is hooked
+// into runits ./finish file (http://smarden.org/runit/runsv.8.html)
+func (e EnvironmentExtractor) WriteFinish(exitCode int, exitStatus int) error {
+	finish, err := e.constructFinishFromEnvironment(exitCode, exitStatus)
+	if err != nil {
+		return err
+	}
+
+	logger := logging.DefaultLogger
```
Comment (anthonybishopric) [inline pkg/preparer/podprocess/env_extractor.go]: Make this a field

File: pkg/preparer/setup.go [hunk 2]
```diff
@@ -397,6 +397,7 @@ func New(preparerConfig *PreparerConfig, logger logging.Logger) (*Preparer, erro
 		}
 
 		finishExec = preparerConfig.PodProcessReporterConfig.FinishExec()
+
```
Comment (anthonybishopric) [inline pkg/preparer/setup.go]: delete

**Comment by mpuncel** [comment]
feedback addressed, will merge on green build

**Pull request merged**